crates/router/src/core/payments/access_token.rs

```diff
@@ -130,6 +137,25 @@ pub async fn add_access_token<
     }
 }
 
+pub fn is_new_access_token_required(old_access_token: Option<&types::AccessToken>) -> bool {
+    match old_access_token {
+        Some(access_token) => {
+            // Access token is present
+            match access_token.refresh_token_created_at {
+                // If access_token is present along with created_at, then the current time should not exceed the expiration time
+                Some(created_at) => {
+                    let now = time::OffsetDateTime::now_utc().unix_timestamp();
+                    now > (created_at + access_token.expires)
```

Comment:
We need to add a extra offset, so that if the toke expires at 1:00, with offset of 5mins we refresh on 12.55 itself.